### DISCUSSION TITLE
How to embed python runtime into my app

### DISCUSSION THREAD

**Discussion by oovm** [RustPython/RustPython#3805] [Q&A] [answered]
I need a custom python runtime without stdlib, I saw an example of this:

https://github.com/RustPython/RustPython/blob/main/examples/hello_embed.rs

but the dependence `rustpython_vm` seems yanked: https://crates.io/crates/rustpython-vm

How can I do the embed today?

I also don't need `freeze-stdlib`, can I set it directly to `utf-8` here?

https://github.com/RustPython/RustPython/blob/60598b699ef06ab22b981301916cde84ce7b2dd2/vm/src/vm/mod.rs#L205-L215

**Comment by youknowone** [comment, marked as answer]
Try to use latest main branch instead of published version.

Even without stdlib, you have to embed a few python files. Skipping stdlib is possible but builtin-modules are necessary. https://docs.python.org/3/library/sys.html#sys.builtin_module_names

**Reply by Rafael-Conde** [reply]
Hello there! Would you mind to help me too? 

I created a new Rust binary project with cargo new and copy pasted the `hello_embed.rs` into the my `main.rs` file. In the `Cargo.toml` file I added

```rust
# ... Boilerplate cargo.toml stuff and then
[dependencies]
rustpython = {git = "http://github.com/rustpython/rustpython.git", features = ["freeze-stdlib"]}
```

I believe I'm using the lastest version of this repo, but yet I still get an error where it can't find `external crate rustpython_vm`. I saw that `rustpython_vm` is added as a dependency to Rustpython and not a member of the workspace, so because of that I don't know how to separate them.

Is the problem in the way I added rustpython? 

Any pointers or help is appreciated! Happy Holidays! 

PS: I typed the url from what I remembered but in reality I copy pasted from the GitHub copy button

**Reply by DimitrisJim** [reply]
Just renaming it should do it I believe:

```
rustpython-vm = {git = "http://github.com/rustpython/rustpython.git", features = ["freeze-stdlib"]}
```

**Reply by Rafael-Conde** [reply]
Appreciated!! it worked!! thank you!
